content/resources/crownlabs_ssh/crownlabs_ssh.md

```diff
@@ -0,0 +1,94 @@
++++
+fragment = "content"
+#disabled = true
+date = "2021-11-12"
+weight = 100
+# title = "Connecting to CrownLabs with SSH"
++++
+
+## Introduction
+
+**CrownLabs** allows users to access their VMs through a local SSH client (i.e., executed on their local machines).
+
+In this guide you are going to learn how to generate a SSH key pair on your operating systems (Windows, Linux, MacOS), how to load it on **CrownLabs** and access your **CrownLabs**' VM using SSH from a local terminal.
+
+**Note that the **generation** of the **SSH key pair** and the upload of the **SSH public key** does not have to be repeated each time you create a new **VM**, this process only needs to be done once per device.**
+
+## How to generate an SSH key pair
+
+1. Open a terminal and run the following command to create a [**RSA**](<https://en.wikipedia.org/wiki/RSA_(cryptosystem)>) key pair :
+
+   ```
+   ssh-keygen
+   ```
+
+   You can use the `-b` option to specify the length (bit size) of the key, as shown in the following example:
+
+   ```
+   ssh-keygen -b 2048
+   ```
+
+2. The command will prompt you to enter the path to the file where you want to save the key. A default path and file name similar to `/home/user_name/.ssh/id_rsa` (`C:\Users\username\.ssh\id_rsa` on **windows**) are suggested in parentheses. To accept the default path and file name (_suggested_), press **Enter**. Otherwise, enter the required path and file name, and then press **Enter**. Beware that from now on, this tutorial will assume you accepted the default settings.
+
+3. A password will be asked. You are free to insert one or to ignore it.
+
+Now you will find in the previously chosen path (`$HOME/.ssh` or `C:\Users\%USERPROFILE%\.ssh` if you kept the default one) a file with a **.pub** extension (`id_rsa.pub` as default). This is the file containing your **SSH public key**.\
+Note that the directory contains a file called `id_rsa` (without **.pub**), it is the **SSH private key** which you must not share with others.\
+Print the content of your **SSH public key** running `cat $HOME/.ssh/id_rsa.pub` or opening `C:\Users\%USERPROFILE%\.ssh` with an editor in **windows**
+## Upload your key on CrownLabs
+
```

Comment:
I suggest:

## Upload your key on CrownLabs

To use SSH with Crownlabs, copy the contents of your public key file to your Crownlabs account.

You can do this manually or use a script. For example, to copy an RSA key to the clipboard:

 **macOS**:

```bash
 tr -d '\n' < ~/.ssh/id_rsa.pub | pbcopy
```

    **Linux (requires the xclip package)**:

```bash 
xclip -sel clip < ~/.ssh/id_rsa.pub
```

   **Git Bash on Windows**:

```bash
 cat ~/.ssh/id_rsa.pub | clip
```

    Replace `id_rsa.pub` with your filename. For example, use `id_ed25519.pub` for ED25519.

In contrast, you can print the content of your **SSH public key** running `cat $HOME/.ssh/id_rsa.pub` or opening `C:\Users\%USERPROFILE%\.ssh` with an editor in
**windows** and then copy it as well.

Reply:
Personally agree with @cheina97. What @Ralls0 proposed is really cool, but probably overkill for the target audience (if a command for any reason fail -- e.g., a package is missing -- then they would not know what to do next). Would keep this tutorial as simple as possible.

Nonetheless, would discuss this during the meeting today.